homeassistant/components/nuki/__init__.py

```diff
@@ -1,3 +1,151 @@
 """The nuki component."""
 
-DOMAIN = "nuki"
+from datetime import timedelta
+import logging
+
+import async_timeout
+from pynuki import NukiBridge
+from requests.exceptions import RequestException
+import voluptuous as vol
+
+from homeassistant import exceptions
+from homeassistant.config_entries import SOURCE_IMPORT
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_TOKEN
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import (
+    DATA_BRIDGE,
+    DATA_COORDINATOR,
+    DATA_LOCKS,
+    DATA_OPENERS,
+    DEFAULT_PORT,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+    ERROR_STATES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+NUKI_COMPONENTS = ["lock", "binary_sensor"]
+UPDATE_INTERVAL = timedelta(seconds=30)
+
+NUKI_SCHEMA = vol.Schema(
+    vol.All(
+        {
+            vol.Required(CONF_HOST): cv.string,
+            vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+            vol.Required(CONF_TOKEN): cv.string,
+        },
+    )
+)
+
+CONFIG_SCHEMA = vol.Schema(
+    {DOMAIN: vol.Schema(vol.All(cv.ensure_list, [NUKI_SCHEMA]))},
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Set up the Nuki component."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    # import
+    if DOMAIN in config:
+        for conf in config[DOMAIN]:
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    DOMAIN, context={"source": SOURCE_IMPORT}, data=conf
+                )
+            )
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Set up the Nuki entry."""
+
+    bridge = await hass.async_add_executor_job(
+        NukiBridge,
+        entry.data[CONF_HOST],
+        entry.data[CONF_TOKEN],
+        entry.data[CONF_PORT],
+        True,
+        DEFAULT_TIMEOUT,
+    )
+
+    try:
+        locks = await hass.async_add_executor_job(bridge._get_devices, 0)
+        openers = await hass.async_add_executor_job(bridge._get_devices, 2)
```

Comment:
```suggestion
        # pylint: disable=protected-access
        locks = await hass.async_add_executor_job(bridge._get_devices, 0)
        openers = await hass.async_add_executor_job(bridge._get_devices, 2)
```

Reply:
id also recommend moving this into a single method and avoid back to back executor calls